controllers/storagecluster/topology.go

```diff
@@ -38,27 +38,40 @@ func (r *StorageClusterReconciler) getStorageClusterEligibleNodes(sc *ocsv1.Stor
 	return nodes, err
 }
 
-// determineFailureDomain determines the appropriate Ceph failure domain based
+// getFailureDomain returns the failure domain that was determined at the time of node topology reconcilation
+func getFailureDomain(sc *ocsv1.StorageCluster) string {
+	return sc.Status.FailureDomain
+}
+
+// setFailureDomain determines the appropriate Ceph failure domain based
 // on the storage cluster's topology map
-func determineFailureDomain(sc *ocsv1.StorageCluster) string {
+func setFailureDomain(sc *ocsv1.StorageCluster) {
+
+	// We don't change the failure domain after it is determined
 	if sc.Status.FailureDomain != "" {
-		return sc.Status.FailureDomain
+		return
 	}
 
+	// default is rack
+	failureDomain := "rack"
+
+	// But if FlexiableScaling is enabled then we select host as failure domain
+	// as we need +1 scaling
 	if sc.Spec.FlexibleScaling {
-		return "host"
+		failureDomain = "host"
 	}
 
+	// If sufficient zones are available then we select zone as the failure domain
 	topologyMap := sc.Status.NodeTopologies
-	failureDomain := "rack"
 	for label, labelValues := range topologyMap.Labels {
 		if strings.Contains(label, "zone") {
 			if (len(labelValues) >= 2 && arbiterEnabled(sc)) || (len(labelValues) >= 3) {
 				failureDomain = "zone"
 			}
 		}
 	}
-	return failureDomain
+
+	sc.Status.FailureDomain = failureDomain
```

Comment:
I think we should update sc here

Reply:
Because of the recent refactoring, all the updates to the sc.Status are handled outside of the reconcilePhases() function. Not updating the sc status to follow the same pattern.